File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int cnt=1;
        int n=nums.length;
        Arrays.sort(nums);
        if(n==1){
            return nums[0];
        }
        for(int i=0;i<n-1;i++){
            if(nums[i]==nums[i+1]){
                cnt++;
                if(cnt>n/2){
                    return nums[i];
                }
            }
        }
        return 0;
    }
}

File: 0169-majority-element/0169-majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        int n=nums.Length;
        int cnt=0;
        int ele=0;
        for(int i=0;i<n;i++){
            if(cnt==0){
                cnt=1;
                ele=nums[i];
            }
            else if(ele==nums[i]){
                cnt++;
            }
            else{
                cnt--;
            }
        }
        int cnt1=0;
        for(int i=0;i<n;i++){
            if(ele==nums[i]){
                cnt1++;
            }
        }
        if(cnt1>(n/2)){
            return ele;
        }
        return -1;
    }
}